sycl/doc/extensions/SPIRV/SPV_INTEL_vector_compute.asciidoc

```diff
@@ -0,0 +1,164 @@
+
+Overview
+--------
+
+This extension adds new execution model, capability, execution modes and decorations
+to support INTEL vector compute specific features in SPIR-V.
+
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_vector_compute"
+----
+
+Modifications to the SPIR-V Specification, Version 1.5, Revision 2, Unified
+---------------------------------------------------------------------------
+
+Modify Section 3.3, Execution Model, add the following row to the Execution Model table:
+
+[cols="1,20,10",options="header",width = "60%"]
+|====
+2+^| Execution Model | Enabling Capabilities
+| 5626 | *VectorComputeKernelINTEL* | *VectorComputeINTEL*
+|====
+
+Modify Section 3.6, Execution Mode, add the following rows to the Execution Mode table:
+
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Execution Mode  | Enabling Capabilities | Extra Operands
+| 5618 | *SharedLocalMemorySizeINTEL* +
+Specifies the size of shared memory required by kernel, in bytes.
+
+'Size' is an unsigned 32-bit integer.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* | <<Literal, 'Literal'>> _Size_
+|====
+--
+
+Modify Section 3.20, Decoration, add the following rows to the Decoration table:
+
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Decoration  | Enabling Capabilities | Extra Operands
+| 5624 | *KernelArgumentTypeINTEL* +
+Apply to a function parameter to provide information about high level language argument type.
+
+_Argument Type_ is an unsigned 32-bit integer.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* | <<Literal, 'Literal'>> _Argument Type_
+| 5625 | *KernelArgumentDescINTEL* +
+Apply to a function parameter to provide a string description of the argument.
+
+'Description' is a literal string.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* | <<Literal, 'Literal'>> _Description_
+| 5627 | *StackCallINTEL* +
+Apply to a function definition to specify that this function is a stack call function.
+
+See the client API specification for more detail.
+| *VectorComputeINTEL* |
+|====
+--
+
+Modify Section 3.31, Capability, add the following row the 'Capability' table:
+--
+[cols="1,20,10,10",options="header",width = "80%"]
+|====
+  2+^| Capability      |     Implicitly Declares    | Enabled by Extension
+| 5617 | *VectorComputeINTEL* +
+Enables the use of *VectorComputeKernelINTEL* execution model, *SharedLocalMemorySizeINTEL* and *NamedBarrierCountINTEL* execution modes and *KernelArgumentTypeINTEL*, *KernelArgumentDescINTEL* and *StackCallINTEL* decorations.
+
+See the client API specification for more detail.
+|| *SPV_INTEL_vector_compute*
+|====
+--
+
+Issues
+------
+
+1) Which, if any, capabilities should VectorComputeINTEL implicitly enable?
+
+
+Discussion:
+
+Kernel, new float controls?
```

Comment:
Should it be removed? It doesn't create any meaningful starting point for discussion, plus some float controls are documented by a separate extension

Reply:
removed